crates/wast/src/ast/types.rs

```diff
@@ -246,22 +252,102 @@ impl<'a> Parse<'a> for FunctionType<'a> {
     }
 }
 
+/// A struct type with fields.
+#[derive(Clone, Debug)]
+pub struct StructType<'a> {
+    /// The fields of the struct
+    pub fields: Vec<StructField<'a>>,
+}
+
+impl<'a> StructType<'a> {
+    fn finish_parse(&mut self, parser: Parser<'a>) -> Result<()> {
```

Comment:
It's fine to inline this into `parse` below, I think the only reason this was split off elsewhere was that it was called from a few locations.